Unable to run `make generate`

I'm trying to update the `heroku.go` with the current latest schema but I'm running into the following error:
```shell
cd v5 && ../script/generate
+ : 1
+ getopts hnv: opt
+ API_VERSION=3
+ CLIENT_VERSION=v5
+ SCHEMA_FILE=v5/schema.json
+ GENERATED_FILE=v5/heroku.go
+++ dirname ../script/generate
++ cd ../script/..
++ pwd
+ NEW_DIR=/dev_exclusions/heroku-go
+ cd /dev_exclusions/heroku-go
+ '[' 1 -eq 1 ']'
+ curl -so v5/schema.json https://api.heroku.com/schema -H 'Accept: application/vnd.heroku+json; version=3'
+ go get -u github.com/interagent/schematic/cmd/schematic
go: finding github.com/interagent/schematic/cmd/schematic latest
go: finding github.com/interagent/schematic/cmd latest
go: finding github.com/interagent/schematic latest
+ schematic v5/schema.json
Error processing schema element:
    {
      "description": "certificate provided by this endpoint",
      "properties": {
        "ca_signed?": {
          "readOnly": true,
          "type": [
            "boolean"
          ]
        },
        "cert_domains": {
          "readOnly": true,
          "type": [
            "array"
          ]
        },
        "expires_at": {
          "readOnly": true,
          "format": "date-time",
          "type": [
            "string"
          ]
        },
        "id": {
          "description": "unique identifier of this SSL certificate",
          "readOnly": true,
          "example": "01234567-89ab-cdef-0123-456789abcdef",
          "format": "uuid",
          "type": [
            "string"
          ]
        },
        "issuer": {
          "readOnly": true,
          "type": [
            "string"
          ]
        },
        "self_signed?": {
          "readOnly": true,
          "type": [
            "boolean"
          ]
        },
        "starts_at": {
          "readOnly": true,
          "format": "date-time",
          "type": [
            "string"
          ]
        },
        "subject": {
          "readOnly": true,
          "type": [
            "string"
          ]
        }
      }
    }

Failed with: unknown type <nil>
make: *** [generate] Error 1
```

This has happened to me in the past and if I recall correctly, the API schema needed to be updated.

@bernerdschaefer @mars Could one of you check if the schema is correct?

--- a/v5/heroku.go
+++ b/v5/heroku.go
@@ -212,9 +212,11 @@ func String(v string) *string {
 // An account represents an individual signed up to use the Heroku
 // platform.
 type Account struct {
-	AllowTracking       bool      `json:"allow_tracking" url:"allow_tracking,key"` // whether to allow third party web activity tracking
-	Beta                bool      `json:"beta" url:"beta,key"`                     // whether allowed to utilize beta Heroku features
-	CreatedAt           time.Time `json:"created_at" url:"created_at,key"`         // when account was created
+	AcknowledgedMsa     bool       `json:"acknowledged_msa" url:"acknowledged_msa,key"`       // whether account has acknowledged the MSA terms of service
+	AcknowledgedMsaAt   *time.Time `json:"acknowledged_msa_at" url:"acknowledged_msa_at,key"` // when account has acknowledged the MSA terms of service
+	AllowTracking       bool       `json:"allow_tracking" url:"allow_tracking,key"`           // whether to allow third party web activity tracking
+	Beta                bool       `json:"beta" url:"beta,key"`                               // whether allowed to utilize beta Heroku features
+	CreatedAt           time.Time  `json:"created_at" url:"created_at,key"`                   // when account was created
 	DefaultOrganization *struct {
 		ID   string `json:"id" url:"id,key"`     // unique identifier of team
 		Name string `json:"name" url:"name,key"` // unique name of team
@@ -241,6 +243,10 @@ type Account struct {
 			Name string `json:"name" url:"name,key"` // unique name of team
 		} `json:"team" url:"team,key"`
 	} `json:"identity_provider" url:"identity_provider,key"` // Identity Provider details for federated users.
+	ItalianCustomerTerms *string `json:"italian_customer_terms" url:"italian_customer_terms,key"` // whether account has acknowledged the Italian customer terms of
+	// service
+	ItalianPartnerTerms *string `json:"italian_partner_terms" url:"italian_partner_terms,key"` // whether account has acknowledged the Italian provider terms of
+	// service
 	LastLogin               *time.Time `json:"last_login" url:"last_login,key"`                               // when account last authorized with Heroku
 	Name                    *string    `json:"name" url:"name,key"`                                           // full name of the account owner
 	SmsNumber               *string    `json:"sms_number" url:"sms_number,key"`                               // SMS number of account
@@ -466,9 +472,9 @@ func (s *Service) AddOnResolution(ctx context.Context, o AddOnResolutionOpts) (A
 }
 
 // Add-on Actions are lifecycle operations for add-on provisioning and
-// deprovisioning. They allow whitelisted add-on providers to
-// (de)provision add-ons in the background and then report back when
-// (de)provisioning is complete.
+// deprovisioning. They allow add-on providers to (de)provision add-ons
+// in the background and then report back when (de)provisioning is
+// complete.
 type AddOnAction struct{}
 type AddOnActionProvisionResult struct {
 	Actions      []struct{} `json:"actions" url:"actions,key"` // provider actions for this specific add-on
@@ -1043,6 +1049,45 @@ func (s *Service) AddOnWebhookEventList(ctx context.Context, addOnIdentity strin
 	return addOnWebhookEvent, s.Get(ctx, &addOnWebhookEvent, fmt.Sprintf("/addons/%v/webhook-events", addOnIdentity), nil, lr)
 }
 
+// Entities that have been allowed to be used by a Team
+type AllowedAddOnService struct {
+	AddedAt time.Time `json:"added_at" url:"added_at,key"` // when the add-on service was allowed
+	AddedBy struct {
+		Email string `json:"email" url:"email,key"` // unique email address of account
+		ID    string `json:"id" url:"id,key"`       // unique identifier of an account
+	} `json:"added_by" url:"added_by,key"` // the user which allowed the add-on service
+	AddonService struct {
+		HumanName string `json:"human_name" url:"human_name,key"` // human-readable name of the add-on service provider
+		ID        string `json:"id" url:"id,key"`                 // unique identifier of this add-on-service
+		Name      string `json:"name" url:"name,key"`             // unique name of this add-on-service
+	} `json:"addon_service" url:"addon_service,key"` // the add-on service allowed for use
+	ID string `json:"id" url:"id,key"` // unique identifier for this allowed add-on service record
+}
+type AllowedAddOnServiceListByTeamResult []AllowedAddOnService
+
+// List all allowed add-on services for a team
+func (s *Service) AllowedAddOnServiceListByTeam(ctx context.Context, teamIdentity string, lr *ListRange) (AllowedAddOnServiceListByTeamResult, error) {
+	var allowedAddOnService AllowedAddOnServiceListByTeamResult
+	return allowedAddOnService, s.Get(ctx, &allowedAddOnService, fmt.Sprintf("/teams/%v/allowed-addon-services", teamIdentity), nil, lr)
+}
+
+type AllowedAddOnServiceCreateByTeamOpts struct {
+	AddonService *string `json:"addon_service,omitempty" url:"addon_service,omitempty,key"` // name of the add-on service to allow
+}
+type AllowedAddOnServiceCreateByTeamResult []AllowedAddOnService
+
+// Allow an Add-on Service
+func (s *Service) AllowedAddOnServiceCreateByTeam(ctx context.Context, teamIdentity string, o AllowedAddOnServiceCreateByTeamOpts) (AllowedAddOnServiceCreateByTeamResult, error) {
+	var allowedAddOnService AllowedAddOnServiceCreateByTeamResult
+	return allowedAddOnService, s.Post(ctx, &allowedAddOnService, fmt.Sprintf("/teams/%v/allowed-addon-services", teamIdentity), o)
+}
+
+// Remove an allowed add-on service
+func (s *Service) AllowedAddOnServiceDeleteByTeam(ctx context.Context, teamIdentity string, allowedAddOnServiceIdentity string) (*AllowedAddOnService, error) {
+	var allowedAddOnService AllowedAddOnService
+	return &allowedAddOnService, s.Delete(ctx, &allowedAddOnService, fmt.Sprintf("/teams/%v/allowed-addon-services/%v", teamIdentity, allowedAddOnServiceIdentity))
+}
+
 // An app represents the program that you would like to deploy and run
 // on Heroku.
 type App struct {
@@ -1571,6 +1616,44 @@ func (s *Service) ArchiveList(ctx context.Context, enterpriseAccountIdentity str
 	return &archive, s.Get(ctx, &archive, fmt.Sprintf("/enterprise-accounts/%v/archives", enterpriseAccountIdentity), nil, lr)
 }
 
+// An audit trail event represents some action on the platform
+type AuditTrailEvent struct {
+	Action string `json:"action" url:"action,key"` // action for the event
+	Actor  struct {
+		Email string `json:"email" url:"email,key"`
+		ID    string `json:"id" url:"id,key"`
+	} `json:"actor" url:"actor,key"` // user who caused event
+	App struct {
+		ID   string `json:"id" url:"id,key"`
+		Name string `json:"name" url:"name,key"`
+	} `json:"app" url:"app,key"` // app upon which event took place
+	CreatedAt         time.Time `json:"created_at" url:"created_at,key"` // when event was created
+	Data              struct{}  `json:"data" url:"data,key"`             // data specific to the event
+	EnterpriseAccount struct {
+		ID   string `json:"id" url:"id,key"`
+		Name string `json:"name" url:"name,key"`
+	} `json:"enterprise_account" url:"enterprise_account,key"` // enterprise account on which the event happened
+	ID    string `json:"id" url:"id,key"` // unique identifier of event
+	Owner struct {
+		Email string `json:"email" url:"email,key"`
+		ID    string `json:"id" url:"id,key"`
+	} `json:"owner" url:"owner,key"` // owner of the app targeted by the event
+	Request struct {
+		IPAddress string `json:"ip_address" url:"ip_address,key"`
+	} `json:"request" url:"request,key"` // information about where the action was triggered
+	Team struct {
+		ID   string `json:"id" url:"id,key"`
+		Name string `json:"name" url:"name,key"`
+	} `json:"team" url:"team,key"` // team on which the event happened
+	Type string `json:"type" url:"type,key"` // type of event
+}
+
+// List existing events.
+func (s *Service) AuditTrailEventList(ctx context.Context, enterpriseAccountIdentity string, lr *ListRange) (*AuditTrailEvent, error) {
+	var auditTrailEvent AuditTrailEvent
+	return &auditTrailEvent, s.Get(ctx, &auditTrailEvent, fmt.Sprintf("/enterprise-accounts/%v/events", enterpriseAccountIdentity), nil, lr)
+}
+
 // A build represents the process of transforming a code tarball into a
 // slug
 type Build struct {
@@ -1806,22 +1889,51 @@ type Domain struct {
 		ID   string `json:"id" url:"id,key"`     // unique identifier of app
 		Name string `json:"name" url:"name,key"` // unique name of app
 	} `json:"app" url:"app,key"` // app that owns the domain
-	CName     *string   `json:"cname" url:"cname,key"`           // canonical name record, the address to point a domain at
-	CreatedAt time.Time `json:"created_at" url:"created_at,key"` // when domain was created
-	Hostname  string    `json:"hostname" url:"hostname,key"`     // full hostname
-	ID        string    `json:"id" url:"id,key"`                 // unique identifier of this domain
-	Kind      string    `json:"kind" url:"kind,key"`             // type of domain name
+	CName       *string   `json:"cname" url:"cname,key"`           // canonical name record, the address to point a domain at
+	CreatedAt   time.Time `json:"created_at" url:"created_at,key"` // when domain was created
+	Hostname    string    `json:"hostname" url:"hostname,key"`     // full hostname
+	ID          string    `json:"id" url:"id,key"`                 // unique identifier of this domain
+	Kind        string    `json:"kind" url:"kind,key"`             // type of domain name
+	SniEndpoint *struct {
+		ID   string `json:"id" url:"id,key"`     // unique identifier of this SNI endpoint
+		Name string `json:"name" url:"name,key"` // unique name for SNI endpoint
+	} `json:"sni_endpoint" url:"sni_endpoint,key"` // sni endpoint the domain is associated with
 	Status    string    `json:"status" url:"status,key"`         // status of this record's cname
 	UpdatedAt time.Time `json:"updated_at" url:"updated_at,key"` // when domain was updated
 }
-type DomainCreateOpts struct {
+type DomainCreateDeprecatedOpts struct {
 	Hostname string `json:"hostname" url:"hostname,key"` // full hostname
+}
+
+// Create a new domain. Deprecated in favor of this same endpoint, but
+// with a new required attribute of `sni_endpoint`. During the
+// transitional phase sni_endpoint can be omitted entirely (current
+// behavior), can be a valid id, or can be null which will skip
+// auto-association.
+func (s *Service) DomainCreateDeprecated(ctx context.Context, appIdentity string, o DomainCreateDeprecatedOpts) (*Domain, error) {
+	var domain Domain
+	return &domain, s.Post(ctx, &domain, fmt.Sprintf("/apps/%v/domains", appIdentity), o)
+}
+
+type DomainCreateOpts struct {
+	Hostname    string  `json:"hostname" url:"hostname,key"`         // full hostname
+	SniEndpoint *string `json:"sni_endpoint" url:"sni_endpoint,key"` // null or unique identifier or name for SNI endpoint
 }
 
 // Create a new domain.
 func (s *Service) DomainCreate(ctx context.Context, appIdentity string, o DomainCreateOpts) (*Domain, error) {
 	var domain Domain
 	return &domain, s.Post(ctx, &domain, fmt.Sprintf("/apps/%v/domains", appIdentity), o)
+}
+
+type DomainUpdateOpts struct {
+	SniEndpoint *string `json:"sni_endpoint" url:"sni_endpoint,key"` // null or unique identifier or name for SNI endpoint
+}
+
+// Associate an SNI endpoint
+func (s *Service) DomainUpdate(ctx context.Context, appIdentity string, domainIdentity string, o DomainUpdateOpts) (*Domain, error) {
+	var domain Domain
+	return &domain, s.Patch(ctx, &domain, fmt.Sprintf("/apps/%v/domains/%v", appIdentity, domainIdentity), o)
 }
 
 // Delete an existing domain
@@ -2140,44 +2252,6 @@ type EnterpriseAccountUsageMonthlyInfoResult []EnterpriseAccountUsageMonthly
 func (s *Service) EnterpriseAccountUsageMonthlyInfo(ctx context.Context, enterpriseAccountID string, lr *ListRange) (EnterpriseAccountUsageMonthlyInfoResult, error) {
 	var enterpriseAccountUsageMonthly EnterpriseAccountUsageMonthlyInfoResult
 	return enterpriseAccountUsageMonthly, s.Get(ctx, &enterpriseAccountUsageMonthly, fmt.Sprintf("/enterprise-accounts/%v/usage/monthly", enterpriseAccountID), nil, lr)
-}
-
-// An audit trail event represents some action on the platform
-type Event struct {
-	Action string `json:"action" url:"action,key"` // action for the event
-	Actor  struct {
-		Email string `json:"email" url:"email,key"`
-		ID    string `json:"id" url:"id,key"`
-	} `json:"actor" url:"actor,key"` // user who caused event
-	App struct {
-		ID   string `json:"id" url:"id,key"`
-		Name string `json:"name" url:"name,key"`
-	} `json:"app" url:"app,key"` // app upon which event took place
-	CreatedAt         time.Time `json:"created_at" url:"created_at,key"` // when event was created
-	Data              struct{}  `json:"data" url:"data,key"`             // data specific to the event
-	EnterpriseAccount struct {
-		ID   string `json:"id" url:"id,key"`
-		Name string `json:"name" url:"name,key"`
-	} `json:"enterprise_account" url:"enterprise_account,key"` // enterprise account on which the event happened
-	ID    string `json:"id" url:"id,key"` // unique identifier of event
-	Owner struct {
-		Email string `json:"email" url:"email,key"`
-		ID    string `json:"id" url:"id,key"`
-	} `json:"owner" url:"owner,key"` // owner of the app targeted by the event
-	Request struct {
-		IPAddress string `json:"ip_address" url:"ip_address,key"`
-	} `json:"request" url:"request,key"` // information about where the action was triggered
-	Team struct {
-		ID   string `json:"id" url:"id,key"`
-		Name string `json:"name" url:"name,key"`
-	} `json:"team" url:"team,key"` // team on which the event happened
-	Type string `json:"type" url:"type,key"` // type of event
-}
-
-// List existing events.
-func (s *Service) EventList(ctx context.Context, enterpriseAccountIdentity string, lr *ListRange) (*Event, error) {
-	var event Event
-	return &event, s.Get(ctx, &event, fmt.Sprintf("/enterprise-accounts/%v/events", enterpriseAccountIdentity), nil, lr)
 }
 
 // Filters are special endpoints to allow for API consumers to specify a
@@ -3531,8 +3605,7 @@ type ReviewApp struct {
 	Pipeline struct {
 		ID string `json:"id" url:"id,key"` // unique identifier of pipeline
 	} `json:"pipeline" url:"pipeline,key"` // the pipeline which this review app belongs to
-	PrNumber *int `json:"pr_number" url:"pr_number,key"` // GitHub Pull Request number if the Review app was created
-	// automatically
+	PrNumber  *int      `json:"pr_number" url:"pr_number,key"`     // pull request number the review app is built for
 	Status    string    `json:"status" url:"status,key"`           // current state of the review app
 	UpdatedAt time.Time `json:"updated_at" url:"updated_at,key"`   // when review app was updated
 	WaitForCi bool      `json:"wait_for_ci" url:"wait_for_ci,key"` // wait for ci before building the app
@@ -3542,6 +3615,7 @@ type ReviewAppCreateOpts struct {
 	Environment map[string]*string `json:"environment,omitempty" url:"environment,omitempty,key"`   // hash of config vars
 	ForkRepoID  *int               `json:"fork_repo_id,omitempty" url:"fork_repo_id,omitempty,key"` // repository id of the fork the branch resides in
 	Pipeline    string             `json:"pipeline" url:"pipeline,key"`                             // unique identifier of pipeline
+	PrNumber    *int               `json:"pr_number,omitempty" url:"pr_number,omitempty,key"`       // pull request number the review app is built for
 	SourceBlob  struct {
 		URL *string `json:"url,omitempty" url:"url,omitempty,key"` // URL where gzipped tar archive of source code for build was
 		// downloaded.
@@ -3935,6 +4009,33 @@ func (s *Service) SpaceNATInfo(ctx context.Context, spaceIdentity string) (*Spac
 	return &spaceNAT, s.Get(ctx, &spaceNAT, fmt.Sprintf("/spaces/%v/nat", spaceIdentity), nil, nil)
 }
 
+// Space Topology provides you with a mechanism for viewing all the
+// running dynos, formations and applications for a space. This is the
+// same data thats used to power our DNS Service Discovery.
+type SpaceTopology struct {
+	Apps []struct {
+		Domains   []interface{} `json:"domains" url:"domains,key"`
+		Formation []struct {
+			Dynos []struct {
+				Hostname  string `json:"hostname" url:"hostname,key"`     // localspace hostname of resource
+				ID        string `json:"id" url:"id,key"`                 // unique identifier of this dyno
+				Number    int    `json:"number" url:"number,key"`         // process number, e.g. 1 in web.1
+				PrivateIP string `json:"private_ip" url:"private_ip,key"` // RFC1918 Address of Dyno
+			} `json:"dynos" url:"dynos,key"` // Current dynos for application
+			ID          string `json:"id" url:"id,key"`                     // unique identifier of this process type
+			ProcessType string `json:"process_type" url:"process_type,key"` // Name of process type
+		} `json:"formation" url:"formation,key"` // formations for application
+		ID string `json:"id" url:"id,key"` // unique identifier of app
+	} `json:"apps" url:"apps,key"` // The apps within this space
+	Version int `json:"version" url:"version,key"` // version of the space topology payload
+}
+
+// Current space topology
+func (s *Service) SpaceTopologyTopology(ctx context.Context, spaceIdentity string) (*SpaceTopology, error) {
+	var spaceTopology SpaceTopology
+	return &spaceTopology, s.Get(ctx, &spaceTopology, fmt.Sprintf("/spaces/%v/topology", spaceIdentity), nil, nil)
+}
+
 // Transfer spaces between enterprise teams with the same Enterprise
 // Account.
 type SpaceTransfer struct{}
@@ -3984,9 +4085,21 @@ type SSLEndpoint struct {
 	CertificateChain string    `json:"certificate_chain" url:"certificate_chain,key"` // raw contents of the public certificate chain (eg: .crt or .pem file)
 	CName            string    `json:"cname" url:"cname,key"`                         // canonical name record, the address to point a domain at
 	CreatedAt        time.Time `json:"created_at" url:"created_at,key"`               // when endpoint was created
+	DisplayName      string    `json:"display_name" url:"display_name,key"`           // unique name for SSL endpoint
+	Domains          []string  `json:"domains" url:"domains,key"`                     // domains associated with this endpoint
 	ID               string    `json:"id" url:"id,key"`                               // unique identifier of this SSL endpoint
 	Name             string    `json:"name" url:"name,key"`                           // unique name for SSL endpoint
-	UpdatedAt        time.Time `json:"updated_at" url:"updated_at,key"`               // when endpoint was updated
+	SSLCert          struct {
+		IsCaSigned   bool          `json:"ca_signed?" url:"ca_signed?,key"`
+		CertDomains  []interface{} `json:"cert_domains" url:"cert_domains,key"`
+		ExpiresAt    time.Time     `json:"expires_at" url:"expires_at,key"`
+		ID           string        `json:"id" url:"id,key"` // unique identifier of this SSL certificate
+		Issuer       string        `json:"issuer" url:"issuer,key"`
+		IsSelfSigned bool          `json:"self_signed?" url:"self_signed?,key"`
+		StartsAt     time.Time     `json:"starts_at" url:"starts_at,key"`
+		Subject      string        `json:"subject" url:"subject,key"`
+	} `json:"ssl_cert" url:"ssl_cert,key"` // certificate provided by this endpoint
+	UpdatedAt time.Time `json:"updated_at" url:"updated_at,key"` // when endpoint was updated
 }
 type SSLEndpointCreateOpts struct {
 	CertificateChain string `json:"certificate_chain" url:"certificate_chain,key"`       // raw contents of the public certificate chain (eg: .crt or .pem file)
@@ -4657,6 +4770,8 @@ func (s *Service) TeamMemberListByMember(ctx context.Context, teamIdentity strin
 
 // Tracks a Team's Preferences
 type TeamPreferences struct {
+	AddonsControls *bool `json:"addons-controls" url:"addons-controls,key"` // Whether add-on service rules should be applied to add-on
+	// installations
 	DefaultPermission   *string `json:"default-permission" url:"default-permission,key"`     // The default permission used when adding new members to the team
 	WhitelistingEnabled *bool   `json:"whitelisting-enabled" url:"whitelisting-enabled,key"` // Whether whitelisting rules should be applied to add-on installations
 }
@@ -4668,6 +4783,8 @@ func (s *Service) TeamPreferencesList(ctx context.Context, teamPreferencesIdenti
 }
 
 type TeamPreferencesUpdateOpts struct {
+	AddonsControls *bool `json:"addons-controls,omitempty" url:"addons-controls,omitempty,key"` // Whether add-on service rules should be applied to add-on
+	// installations
 	WhitelistingEnabled *bool `json:"whitelisting-enabled,omitempty" url:"whitelisting-enabled,omitempty,key"` // Whether whitelisting rules should be applied to add-on installations
 }
 
@@ -4853,9 +4970,11 @@ type TestRun struct {
 	Status        string    `json:"status" url:"status,key"`                   // current state of the test run
 	UpdatedAt     time.Time `json:"updated_at" url:"updated_at,key"`           // when test-run was updated
 	User          struct {
-		AllowTracking       bool      `json:"allow_tracking" url:"allow_tracking,key"` // whether to allow third party web activity tracking
-		Beta                bool      `json:"beta" url:"beta,key"`                     // whether allowed to utilize beta Heroku features
-		CreatedAt           time.Time `json:"created_at" url:"created_at,key"`         // when account was created
+		AcknowledgedMsa     bool       `json:"acknowledged_msa" url:"acknowledged_msa,key"`       // whether account has acknowledged the MSA terms of service
+		AcknowledgedMsaAt   *time.Time `json:"acknowledged_msa_at" url:"acknowledged_msa_at,key"` // when account has acknowledged the MSA terms of service
+		AllowTracking       bool       `json:"allow_tracking" url:"allow_tracking,key"`           // whether to allow third party web activity tracking
+		Beta                bool       `json:"beta" url:"beta,key"`                               // whether allowed to utilize beta Heroku features
+		CreatedAt           time.Time  `json:"created_at" url:"created_at,key"`                   // when account was created
 		DefaultOrganization *struct {
 			ID   string `json:"id" url:"id,key"`     // unique identifier of team
 			Name string `json:"name" url:"name,key"` // unique name of team
@@ -4882,6 +5001,10 @@ type TestRun struct {
 				Name string `json:"name" url:"name,key"` // unique name of team
 			} `json:"team" url:"team,key"`
 		} `json:"identity_provider" url:"identity_provider,key"` // Identity Provider details for federated users.
+		ItalianCustomerTerms *string `json:"italian_customer_terms" url:"italian_customer_terms,key"` // whether account has acknowledged the Italian customer terms of
+		// service
+		ItalianPartnerTerms *string `json:"italian_partner_terms" url:"italian_partner_terms,key"` // whether account has acknowledged the Italian provider terms of
+		// service
 		LastLogin               *time.Time `json:"last_login" url:"last_login,key"`                               // when account last authorized with Heroku
 		Name                    *string    `json:"name" url:"name,key"`                                           // full name of the account owner
 		SmsNumber               *string    `json:"sms_number" url:"sms_number,key"`                               // SMS number of account

--- a/v5/schema.json
+++ b/v5/schema.json
@@ -359,6 +359,42 @@
           "type": [
             "boolean"
           ]
+        },
+        "acknowledged_msa": {
+          "description": "whether account has acknowledged the MSA terms of service",
+          "example": false,
+          "readOnly": true,
+          "type": [
+            "boolean"
+          ]
+        },
+        "acknowledged_msa_at": {
+          "description": "when account has acknowledged the MSA terms of service",
+          "example": "2012-01-01T12:00:00Z",
+          "format": "date-time",
+          "readOnly": true,
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "italian_customer_terms": {
+          "description": "whether account has acknowledged the Italian customer terms of service",
+          "example": "affirmatively_accepted",
+          "readOnly": true,
+          "type": [
+            "string",
+            "null"
+          ]
+        },
+        "italian_partner_terms": {
+          "description": "whether account has acknowledged the Italian provider terms of service",
+          "example": "affirmatively_accepted",
+          "readOnly": true,
+          "type": [
+            "string",
+            "null"
+          ]
         }
       },
       "links": [
@@ -572,6 +608,18 @@
         "verified": {
           "$ref": "#/definitions/account/definitions/verified"
         },
+        "acknowledged_msa": {
+          "$ref": "#/definitions/account/definitions/acknowledged_msa"
+        },
+        "acknowledged_msa_at": {
+          "$ref": "#/definitions/account/definitions/acknowledged_msa_at"
+        },
+        "italian_customer_terms": {
+          "$ref": "#/definitions/account/definitions/italian_customer_terms"
+        },
+        "italian_partner_terms": {
+          "$ref": "#/definitions/account/definitions/italian_partner_terms"
+        },
         "default_organization": {
           "description": "team selected by default",
           "properties": {
@@ -607,7 +655,7 @@
       }
     },
     "add-on-action": {
-      "description": "Add-on Actions are lifecycle operations for add-on provisioning and deprovisioning. They allow whitelisted add-on providers to (de)provision add-ons in the background and then report back when (de)provisioning is complete.",
+      "description": "Add-on Actions are lifecycle operations for add-on provisioning and deprovisioning. They allow add-on providers to (de)provision add-ons in the background and then report back when (de)provisioning is complete.",
       "$schema": "http://json-schema.org/draft-04/hyper-schema",
       "stability": "development",
       "strictProperties": true,
@@ -2137,6 +2185,157 @@
         },
         "web_url": {
           "$ref": "#/definitions/add-on/definitions/web_url"
+        }
+      }
+    },
+    "allowed-add-on-service": {
+      "description": "Entities that have been allowed to be used by a Team",
+      "$schema": "http://json-schema.org/draft-04/hyper-schema",
+      "stability": "prototype",
+      "strictProperties": true,
+      "title": "Heroku Platform API - Allowed Add-on Service",
+      "type": [
+        "object"
+      ],
+      "definitions": {
+        "added_at": {
+          "description": "when the add-on service was allowed",
+          "example": "2012-01-01T12:00:00Z",
+          "format": "date-time",
+          "readOnly": true,
+          "type": [
+            "string"
+          ]
+        },
+        "added_by": {
+          "description": "the user which allowed the add-on service",
+          "properties": {
+            "email": {
+              "$ref": "#/definitions/account/definitions/email",
+              "type": [
+                "string",
+                "null"
+              ]
+            },
+            "id": {
+              "$ref": "#/definitions/account/definitions/id",
+              "type": [
+                "string",
+                "null"
+              ]
+            }
+          },
+          "readOnly": true,
+          "type": [
+            "object"
+          ]
+        },
+        "addon_service": {
+          "description": "the add-on service allowed for use",
+          "properties": {
+            "id": {
+              "$ref": "#/definitions/add-on-service/definitions/id"
+            },
+            "name": {
+              "$ref": "#/definitions/add-on-service/definitions/name"
+            },
+            "human_name": {
+              "$ref": "#/definitions/add-on-service/definitions/human_name"
+            }
+          },
+          "readOnly": true,
+          "type": [
+            "object"
+          ]
+        },
+        "id": {
+          "description": "unique identifier for this allowed add-on service record",
+          "example": "01234567-89ab-cdef-0123-456789abcdef",
+          "format": "uuid",
+          "readOnly": true,
+          "type": [
+            "string"
+          ]
+        },
+        "identity": {
+          "anyOf": [
+            {
+              "$ref": "#/definitions/allowed-add-on-service/definitions/id"
+            },
+            {
+              "$ref": "#/definitions/add-on-service/definitions/name"
+            }
+          ]
+        }
+      },
+      "links": [
+        {
+          "description": "List all allowed add-on services for a team",
+          "href": "/teams/{(%23%2Fdefinitions%2Fteam%2Fdefinitions%2Fidentity)}/allowed-addon-services",
+          "method": "GET",
+          "rel": "instances",
+          "targetSchema": {
+            "items": {
+              "$ref": "#/definitions/allowed-add-on-service"
+            },
+            "type": [
+              "array"
+            ]
+          },
+          "title": "List By Team"
+        },
+        {
+          "description": "Allow an Add-on Service",
+          "href": "/teams/{(%23%2Fdefinitions%2Fteam%2Fdefinitions%2Fidentity)}/allowed-addon-services",
+          "method": "POST",
+          "rel": "create",
+          "schema": {
+            "type": [
+              "object"
+            ],
+            "properties": {
+              "addon_service": {
+                "description": "name of the add-on service to allow",
+                "example": "heroku-postgresql",
+                "type": [
+                  "string"
+                ]
+              }
+            }
+          },
+          "targetSchema": {
+            "items": {
+              "$ref": "#/definitions/allowed-add-on-service"
+            },
+            "type": [
+              "array"
+            ]
+          },
+          "title": "Create By Team"
+        },
+        {
+          "description": "Remove an allowed add-on service",
+          "href": "/teams/{(%23%2Fdefinitions%2Fteam%2Fdefinitions%2Fidentity)}/allowed-addon-services/{(%23%2Fdefinitions%2Fallowed-add-on-service%2Fdefinitions%2Fidentity)}",
+          "method": "DELETE",
+          "rel": "destroy",
+          "targetSchema": {
+            "$ref": "#/definitions/allowed-add-on-service"
+          },
+          "title": "Delete By Team"
+        }
+      ],
+      "properties": {
+        "added_at": {
+          "$ref": "#/definitions/allowed-add-on-service/definitions/added_at"
+        },
+        "added_by": {
+          "$ref": "#/definitions/allowed-add-on-service/definitions/added_by"
+        },
+        "addon_service": {
+          "$ref": "#/definitions/allowed-add-on-service/definitions/addon_service"
+        },
+        "id": {
+          "$ref": "#/definitions/allowed-add-on-service/definitions/id"
         }
       }
     },
@@ -4028,7 +4227,7 @@
       "$schema": "http://json-schema.org/draft-04/hyper-schema",
       "title": "Heroku Platform API - Audit Trail Archive",
       "description": "An audit trail archive represents a monthly json zipped file containing events",
-      "stability": "development",
+      "stability": "production",
       "strictProperties": true,
       "type": [
         "object"
@@ -4128,6 +4327,222 @@
         },
         "size": {
           "$ref": "#/definitions/archive/definitions/size"
+        }
+      }
+    },
+    "audit-trail-event": {
+      "$schema": "http://json-schema.org/draft-04/hyper-schema",
+      "title": "Heroku Platform API - Audit Trail Event",
+      "description": "An audit trail event represents some action on the platform",
+      "stability": "production",
+      "strictProperties": true,
+      "type": [
+        "object"
+      ],
+      "definitions": {
+        "id": {
+          "description": "unique identifier of event",
+          "readOnly": true,
+          "format": "uuid",
+          "type": [
+            "string"
+          ]
+        },
+        "identity": {
+          "anyOf": [
+            {
+              "$ref": "#/definitions/audit-trail-event/definitions/id"
+            }
+          ]
+        },
+        "type": {
+          "description": "type of event",
+          "readOnly": true,
+          "type": [
+            "string"
+          ]
+        },
+        "action": {
+          "description": "action for the event",
+          "readOnly": true,
+          "type": [
+            "string"
+          ]
+        },
+        "actor": {
+          "description": "user who caused event",
+          "readOnly": true,
+          "type": [
+            "object"
+          ],
+          "properties": {
+            "id": {
+              "format": "uuid",
+              "type": [
+                "string"
+              ]
+            },
+            "email": {
+              "format": "email",
+              "type": [
+                "string"
+              ]
+            }
+          }
+        },
+        "app": {
+          "description": "app upon which event took place",
+          "readOnly": true,
+          "type": [
+            "object"
+          ],
+          "properties": {
+            "id": {
+              "format": "uuid",
+              "type": [
+                "string"
+              ]
+            },
+            "name": {
+              "type": [
+                "string"
+              ]
+            }
+          }
+        },
+        "owner": {
+          "description": "owner of the app targeted by the event",
+          "readOnly": true,
+          "type": [
+            "object"
+          ],
+          "properties": {
+            "id": {
+              "format": "uuid",
+              "type": [
+                "string"
+              ]
+            },
+            "email": {
+              "format": "email",
+              "type": [
+                "string"
+              ]
+            }
+          }
+        },
+        "enterprise_account": {
+          "description": "enterprise account on which the event happened",
+          "readOnly": true,
+          "type": [
+            "object"
+          ],
+          "properties": {
+            "id": {
+              "format": "uuid",
+              "type": [
+                "string"
+              ]
+            },
+            "name": {
+              "type": [
+                "string"
+              ]
+            }
+          }
+        },
+        "team": {
+          "description": "team on which the event happened",
+          "readOnly": true,
+          "type": [
+            "object"
+          ],
+          "properties": {
+            "id": {
+              "format": "uuid",
+              "type": [
+                "string"
+              ]
+            },
+            "name": {
+              "type": [
+                "string"
+              ]
+            }
+          }
+        },
+        "request": {
+          "description": "information about where the action was triggered",
+          "readOnly": true,
+          "type": [
+            "object"
+          ],
+          "properties": {
+            "ip_address": {
+              "format": "ipv4",
+              "type": [
+                "string"
+              ]
+            }
+          }
+        },
+        "data": {
+          "description": "data specific to the event",
+          "readOnly": true,
+          "type": [
+            "object"
+          ]
+        },
+        "created_at": {
+          "description": "when event was created",
+          "format": "date-time",
+          "type": [
+            "string"
+          ]
+        }
+      },
+      "links": [
+        {
+          "description": "List existing events.",
+          "href": "/enterprise-accounts/{(%23%2Fdefinitions%2Fenterprise-account%2Fdefinitions%2Fidentity)}/events",
+          "method": "GET",
+          "rel": "instances",
+          "title": "List"
+        }
+      ],
+      "properties": {
+        "created_at": {
+          "$ref": "#/definitions/audit-trail-event/definitions/created_at"
+        },
+        "id": {
+          "$ref": "#/definitions/audit-trail-event/definitions/id"
+        },
+        "type": {
+          "$ref": "#/definitions/audit-trail-event/definitions/type"
+        },
+        "action": {
+          "$ref": "#/definitions/audit-trail-event/definitions/action"
+        },
+        "actor": {
+          "$ref": "#/definitions/audit-trail-event/definitions/actor"
+        },
+        "app": {
+          "$ref": "#/definitions/audit-trail-event/definitions/app"
+        },
+        "owner": {
+          "$ref": "#/definitions/audit-trail-event/definitions/owner"
+        },
+        "enterprise_account": {
+          "$ref": "#/definitions/audit-trail-event/definitions/enterprise_account"
+        },
+        "team": {
+          "$ref": "#/definitions/audit-trail-event/definitions/team"
+        },
+        "request": {
+          "$ref": "#/definitions/audit-trail-event/definitions/request"
+        },
+        "data": {
+          "$ref": "#/definitions/audit-trail-event/definitions/data"
         }
       }
     },
@@ -5093,11 +5508,19 @@
           "type": [
             "string"
           ]
+        },
+        "sni_endpoint": {
+          "description": "null or unique identifier or name for SNI endpoint",
+          "type": [
+            "null",
+            "string"
+          ]
         }
       },
       "links": [
         {
-          "description": "Create a new domain.",
+          "deactivate_on": "2021-10-31",
+          "description": "Create a new domain. Deprecated in favor of this same endpoint, but with a new required attribute of `sni_endpoint`. During the transitional phase sni_endpoint can be omitted entirely (current behavior), can be a valid id, or can be null which will skip auto-association.",
           "href": "/apps/{(%23%2Fdefinitions%2Fapp%2Fdefinitions%2Fidentity)}/domains",
           "method": "POST",
           "rel": "create",
@@ -5117,7 +5540,57 @@
           "targetSchema": {
             "$ref": "#/definitions/domain"
           },
+          "title": "Create - Deprecated"
+        },
+        {
+          "description": "Create a new domain.",
+          "href": "/apps/{(%23%2Fdefinitions%2Fapp%2Fdefinitions%2Fidentity)}/domains",
+          "method": "POST",
+          "rel": "create",
+          "schema": {
+            "properties": {
+              "hostname": {
+                "$ref": "#/definitions/domain/definitions/hostname"
+              },
+              "sni_endpoint": {
+                "$ref": "#/definitions/domain/definitions/sni_endpoint"
+              }
+            },
+            "required": [
+              "hostname",
+              "sni_endpoint"
+            ],
+            "type": [
+              "object"
+            ]
+          },
+          "targetSchema": {
+            "$ref": "#/definitions/domain"
+          },
           "title": "Create"
+        },
+        {
+          "description": "Associate an SNI endpoint",
+          "href": "/apps/{(%23%2Fdefinitions%2Fapp%2Fdefinitions%2Fidentity)}/domains/{(%23%2Fdefinitions%2Fdomain%2Fdefinitions%2Fidentity)}",
+          "method": "PATCH",
+          "rel": "update",
+          "schema": {
+            "properties": {
+              "sni_endpoint": {
+                "$ref": "#/definitions/domain/definitions/sni_endpoint"
+              }
+            },
+            "required": [
+              "sni_endpoint"
+            ],
+            "type": [
+              "object"
+            ]
+          },
+          "targetSchema": {
+            "$ref": "#/definitions/domain"
+          },
+          "title": "Update"
         },
         {
           "description": "Delete an existing domain",
@@ -5196,6 +5669,21 @@
         },
         "status": {
           "$ref": "#/definitions/domain/definitions/status"
+        },
+        "sni_endpoint": {
+          "description": "sni endpoint the domain is associated with",
+          "properties": {
+            "name": {
+              "$ref": "#/definitions/sni-endpoint/definitions/name"
+            },
+            "id": {
+              "$ref": "#/definitions/sni-endpoint/definitions/id"
+            }
+          },
+          "type": [
+            "null",
+            "object"
+          ]
         }
       }
     },
@@ -6418,222 +6906,6 @@
               "$ref": "#/definitions/identity-provider/definitions/owner"
             }
           }
-        }
-      }
-    },
-    "event": {
-      "$schema": "http://json-schema.org/draft-04/hyper-schema",
-      "title": "Heroku Platform API - Audit Trail Event",
-      "description": "An audit trail event represents some action on the platform",
-      "stability": "development",
-      "strictProperties": true,
-      "type": [
-        "object"
-      ],
-      "definitions": {
-        "id": {
-          "description": "unique identifier of event",
-          "readOnly": true,
-          "format": "uuid",
-          "type": [
-            "string"
-          ]
-        },
-        "identity": {
-          "anyOf": [
-            {
-              "$ref": "#/definitions/event/definitions/id"
-            }
-          ]
-        },
-        "type": {
-          "description": "type of event",
-          "readOnly": true,
-          "type": [
-            "string"
-          ]
-        },
-        "action": {
-          "description": "action for the event",
-          "readOnly": true,
-          "type": [
-            "string"
-          ]
-        },
-        "actor": {
-          "description": "user who caused event",
-          "readOnly": true,
-          "type": [
-            "object"
-          ],
-          "properties": {
-            "id": {
-              "format": "uuid",
-              "type": [
-                "string"
-              ]
-            },
-            "email": {
-              "format": "email",
-              "type": [
-                "string"
-              ]
-            }
-          }
-        },
-        "app": {
-          "description": "app upon which event took place",
-          "readOnly": true,
-          "type": [
-            "object"
-          ],
-          "properties": {
-            "id": {
-              "format": "uuid",
-              "type": [
-                "string"
-              ]
-            },
-            "name": {
-              "type": [
-                "string"
-              ]
-            }
-          }
-        },
-        "owner": {
-          "description": "owner of the app targeted by the event",
-          "readOnly": true,
-          "type": [
-            "object"
-          ],
-          "properties": {
-            "id": {
-              "format": "uuid",
-              "type": [
-                "string"
-              ]
-            },
-            "email": {
-              "format": "email",
-              "type": [
-                "string"
-              ]
-            }
-          }
-        },
-        "enterprise_account": {
-          "description": "enterprise account on which the event happened",
-          "readOnly": true,
-          "type": [
-            "object"
-          ],
-          "properties": {
-            "id": {
-              "format": "uuid",
-              "type": [
-                "string"
-              ]
-            },
-            "name": {
-              "type": [
-                "string"
-              ]
-            }
-          }
-        },
-        "team": {
-          "description": "team on which the event happened",
-          "readOnly": true,
-          "type": [
-            "object"
-          ],
-          "properties": {
-            "id": {
-              "format": "uuid",
-              "type": [
-                "string"
-              ]
-            },
-            "name": {
-              "type": [
-                "string"
-              ]
-            }
-          }
-        },
-        "request": {
-          "description": "information about where the action was triggered",
-          "readOnly": true,
-          "type": [
-            "object"
-          ],
-          "properties": {
-            "ip_address": {
-              "format": "ipv4",
-              "type": [
-                "string"
-              ]
-            }
-          }
-        },
-        "data": {
-          "description": "data specific to the event",
-          "readOnly": true,
-          "type": [
-            "object"
-          ]
-        },
-        "created_at": {
-          "description": "when event was created",
-          "format": "date-time",
-          "type": [
-            "string"
-          ]
-        }
-      },
-      "links": [
-        {
-          "description": "List existing events.",
-          "href": "/enterprise-accounts/{(%23%2Fdefinitions%2Fenterprise-account%2Fdefinitions%2Fidentity)}/events",
-          "method": "GET",
-          "rel": "instances",
-          "title": "List"
-        }
-      ],
-      "properties": {
-        "created_at": {
-          "$ref": "#/definitions/event/definitions/created_at"
-        },
-        "id": {
-          "$ref": "#/definitions/event/definitions/id"
-        },
-        "type": {
-          "$ref": "#/definitions/event/definitions/type"
-        },
-        "action": {
-          "$ref": "#/definitions/event/definitions/action"
-        },
-        "actor": {
-          "$ref": "#/definitions/event/definitions/actor"
-        },
-        "app": {
-          "$ref": "#/definitions/event/definitions/app"
-        },
-        "owner": {
-          "$ref": "#/definitions/event/definitions/owner"
-        },
-        "enterprise_account": {
-          "$ref": "#/definitions/event/definitions/enterprise_account"
-        },
-        "team": {
-          "$ref": "#/definitions/event/definitions/team"
-        },
-        "request": {
-          "$ref": "#/definitions/event/definitions/request"
-        },
-        "data": {
-          "$ref": "#/definitions/event/definitions/data"
         }
       }
     },
@@ -11495,7 +11767,7 @@
           ]
         },
         "pr_number": {
-          "description": "GitHub Pull Request number if the Review app was created automatically",
+          "description": "pull request number the review app is built for",
           "example": 24,
           "readOnly": true,
           "type": [
@@ -11514,6 +11786,9 @@
             "properties": {
               "branch": {
                 "$ref": "#/definitions/review-app/definitions/branch"
+              },
+              "pr_number": {
+                "$ref": "#/definitions/review-app/definitions/pr_number"
               },
               "pipeline": {
                 "$ref": "#/definitions/pipeline/definitions/id"
@@ -12119,7 +12394,7 @@
               "size": 2048,
               "stack": {
                 "id": "01234567-89ab-cdef-0123-456789abcdef",
-                "name": "cedar-14"
+                "name": "heroku-18"
               },
               "updated_at": "2012-01-01T12:00:00Z"
             }
@@ -12742,6 +13017,140 @@
         }
       }
     },
+    "space-topology": {
+      "description": "Space Topology provides you with a mechanism for viewing all the running dynos, formations and applications for a space. This is the same data thats used to power our DNS Service Discovery.",
+      "$schema": "http://json-schema.org/draft-04/hyper-schema",
+      "stability": "prototype",
+      "strictProperties": true,
+      "title": "Heroku Platform API - Space Topology",
+      "type": [
+        "object"
+      ],
+      "definitions": {
+        "version": {
+          "description": "version of the space topology payload",
+          "example": 1,
+          "readOnly": true,
+          "type": [
+            "integer"
+          ]
+        },
+        "dyno": {
+          "description": "A dyno",
+          "properties": {
+            "id": {
+              "$ref": "#/definitions/dyno/definitions/id"
+            },
+            "number": {
+              "description": "process number, e.g. 1 in web.1",
+              "example": 1,
+              "type": [
+                "integer"
+              ]
+            },
+            "private_ip": {
+              "description": "RFC1918 Address of Dyno",
+              "example": "10.0.134.42",
+              "type": [
+                "string"
+              ]
+            },
+            "hostname": {
+              "description": "localspace hostname of resource",
+              "example": "1.example-app-90210.app.localspace",
+              "type": [
+                "string"
+              ]
+            }
+          },
+          "type": [
+            "object"
+          ]
+        },
+        "formation": {
+          "description": "formations for application",
+          "properties": {
+            "id": {
+              "$ref": "#/definitions/formation/definitions/id"
+            },
+            "process_type": {
+              "description": "Name of process type",
+              "example": "web",
+              "type": [
+                "string"
+              ]
+            },
+            "dynos": {
+              "description": "Current dynos for application",
+              "items": {
+                "$ref": "#/definitions/space-topology/definitions/dyno"
+              },
+              "type": [
+                "array"
+              ]
+            }
+          },
+          "type": [
+            "object"
+          ]
+        }
+      },
+      "links": [
+        {
+          "description": "Current space topology",
+          "href": "/spaces/{(%23%2Fdefinitions%2Fspace%2Fdefinitions%2Fidentity)}/topology",
+          "method": "GET",
+          "rel": "self",
+          "targetSchema": {
+            "$ref": "#/definitions/space-topology"
+          },
+          "title": "Topology"
+        }
+      ],
+      "properties": {
+        "version": {
+          "$ref": "#/definitions/space-topology/definitions/version"
+        },
+        "apps": {
+          "description": "The apps within this space",
+          "type": [
+            "array"
+          ],
+          "readOnly": true,
+          "items": {
+            "type": [
+              "object"
+            ],
+            "properties": {
+              "id": {
+                "$ref": "#/definitions/app/definitions/id",
+                "readOnly": true
+              },
+              "domains": {
+                "example": [
+                  "example.com",
+                  "example.net"
+                ],
+                "readOnly": true,
+                "type": [
+                  "array"
+                ]
+              },
+              "formation": {
+                "description": "formations for application",
+                "items": {
+                  "$ref": "#/definitions/space-topology/definitions/formation"
+                },
+                "type": [
+                  "array"
+                ],
+                "readOnly": true
+              }
+            }
+          }
+        }
+      }
+    },
     "space-transfer": {
       "description": "Transfer spaces between enterprise teams with the same Enterprise Account.",
       "$schema": "http://json-schema.org/draft-04/hyper-schema",
@@ -13046,6 +13455,24 @@
         "object"
       ],
       "definitions": {
+        "acm": {
+          "readOnly": true,
+          "type": [
+            "boolean"
+          ]
+        },
+        "ca_signed?": {
+          "readOnly": true,
+          "type": [
+            "boolean"
+          ]
+        },
+        "cert_domains": {
+          "readOnly": true,
+          "type": [
+            "array"
+          ]
+        },
         "certificate_chain": {
           "description": "raw contents of the public certificate chain (eg: .crt or .pem file)",
           "example": "-----BEGIN CERTIFICATE----- ...",
@@ -13071,6 +13498,22 @@
             "string"
           ]
         },
+        "display_name": {
+          "description": "unique name for SSL endpoint",
+          "example": "example",
+          "pattern": "^[a-z][a-z0-9-]{2,29}$",
+          "readOnly": false,
+          "type": [
+            "string"
+          ]
+        },
+        "expires_at": {
+          "readOnly": true,
+          "format": "date-time",
+          "type": [
+            "string"
+          ]
+        },
         "id": {
           "description": "unique identifier of this SSL endpoint",
           "example": "01234567-89ab-cdef-0123-456789abcdef",
@@ -13088,6 +13531,12 @@
             {
               "$ref": "#/definitions/ssl-endpoint/definitions/name"
             }
+          ]
+        },
+        "issuer": {
+          "readOnly": true,
+          "type": [
+            "string"
           ]
         },
         "name": {
@@ -13112,6 +13561,25 @@
           "description": "contents of the private key (eg .key file)",
           "example": "-----BEGIN RSA PRIVATE KEY----- ...",
           "readOnly": false,
+          "type": [
+            "string"
+          ]
+        },
+        "self_signed?": {
+          "readOnly": true,
+          "type": [
+            "boolean"
+          ]
+        },
+        "starts_at": {
+          "readOnly": true,
+          "format": "date-time",
+          "type": [
+            "string"
+          ]
+        },
+        "subject": {
+          "readOnly": true,
           "type": [
             "string"
           ]
@@ -13244,11 +13712,61 @@
         "created_at": {
           "$ref": "#/definitions/ssl-endpoint/definitions/created_at"
         },
+        "display_name": {
+          "$ref": "#/definitions/ssl-endpoint/definitions/display_name"
+        },
+        "domains": {
+          "description": "domains associated with this endpoint",
+          "type": [
+            "array"
+          ],
+          "items": {
+            "$ref": "#/definitions/domain/definitions/id"
+          }
+        },
         "id": {
           "$ref": "#/definitions/ssl-endpoint/definitions/id"
         },
         "name": {
           "$ref": "#/definitions/ssl-endpoint/definitions/name"
+        },
+        "ssl_cert": {
+          "description": "certificate provided by this endpoint",
+          "type": [
+            "object"
+          ],
+          "properties": {
+            "ca_signed?": {
+              "$ref": "#/definitions/ssl-endpoint/definitions/ca_signed?"
+            },
+            "cert_domains": {
+              "$ref": "#/definitions/ssl-endpoint/definitions/cert_domains"
+            },
+            "expires_at": {
+              "$ref": "#/definitions/ssl-endpoint/definitions/expires_at"
+            },
+            "issuer": {
+              "$ref": "#/definitions/ssl-endpoint/definitions/issuer"
+            },
+            "self_signed?": {
+              "$ref": "#/definitions/ssl-endpoint/definitions/self_signed?"
+            },
+            "starts_at": {
+              "$ref": "#/definitions/ssl-endpoint/definitions/starts_at"
+            },
+            "subject": {
+              "$ref": "#/definitions/ssl-endpoint/definitions/subject"
+            },
+            "id": {
+              "description": "unique identifier of this SSL certificate",
+              "example": "01234567-89ab-cdef-0123-456789abcdef",
+              "format": "uuid",
+              "readOnly": true,
+              "type": [
+                "string"
+              ]
+            }
+          }
         },
         "updated_at": {
           "$ref": "#/definitions/ssl-endpoint/definitions/updated_at"
@@ -13303,7 +13821,7 @@
         },
         "name": {
           "description": "unique name of stack",
-          "example": "cedar-14",
+          "example": "heroku-18",
           "readOnly": true,
           "type": [
             "string"
@@ -14863,6 +15381,15 @@
             "boolean",
             "null"
           ]
+        },
+        "addons-controls": {
+          "description": "Whether add-on service rules should be applied to add-on installations",
+          "example": true,
+          "readOnly": false,
+          "type": [
+            "boolean",
+            "null"
+          ]
         }
       },
       "links": [
@@ -14888,6 +15415,9 @@
             "properties": {
               "whitelisting-enabled": {
                 "$ref": "#/definitions/team-preferences/definitions/whitelisting-enabled"
+              },
+              "addons-controls": {
+                "$ref": "#/definitions/team-preferences/definitions/addons-controls"
               }
             }
           },
@@ -14903,6 +15433,9 @@
         },
         "whitelisting-enabled": {
           "$ref": "#/definitions/team-preferences/definitions/whitelisting-enabled"
+        },
+        "addons-controls": {
+          "$ref": "#/definitions/team-preferences/definitions/addons-controls"
         }
       }
     },
@@ -17020,6 +17553,9 @@
     "add-on": {
       "$ref": "#/definitions/add-on"
     },
+    "allowed-add-on-service": {
+      "$ref": "#/definitions/allowed-add-on-service"
+    },
     "app-feature": {
       "$ref": "#/definitions/app-feature"
     },
@@ -17046,6 +17582,9 @@
     },
     "archive": {
       "$ref": "#/definitions/archive"
+    },
+    "audit-trail-event": {
+      "$ref": "#/definitions/audit-trail-event"
     },
     "build": {
       "$ref": "#/definitions/build"
@@ -17082,9 +17621,6 @@
     },
     "enterprise-account": {
       "$ref": "#/definitions/enterprise-account"
-    },
-    "event": {
-      "$ref": "#/definitions/event"
     },
     "filter-apps": {
       "$ref": "#/definitions/filter-apps"
@@ -17205,6 +17741,9 @@
     },
     "space-nat": {
       "$ref": "#/definitions/space-nat"
+    },
+    "space-topology": {
+      "$ref": "#/definitions/space-topology"
     },
     "space-transfer": {
       "$ref": "#/definitions/space-transfer"

